sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py

```diff
@@ -27,6 +31,57 @@
 JSON = MutableMapping[str, Any]  # pylint: disable=unsubscriptable-object
 T = TypeVar("T")
 ClsType = Optional[Callable[[PipelineResponse[HttpRequest, HttpResponse], T, Dict[str, Any]], Any]]
+PollingReturnType_co = TypeVar("PollingReturnType_co", covariant=True)
+_FINISHED = frozenset(["succeeded", "canceled", "failed", "completed"])
+
+
+def _parse_operation_id(operation_location_header):
+    regex = "[^:]+://[^/]+/documentintelligence/.+/([^?/]+)"
+    return re.match(regex, operation_location_header).group(1)
+
+
+class AnalyzeDocumentLROPollingMethod(LROBasePolling):
+    @property
+    def operation_id(self) -> str:
+        """Analyze operation result ID.
+
+        :rtype:str
+        """
+        return _parse_operation_id(self._initial_response.http_response.headers["Operation-Location"])
+
+
+class AnalyzeDocumentLROPoller(LROPoller[PollingReturnType_co]):
+    def polling_method(self) -> AnalyzeDocumentLROPollingMethod:
+        """Return the polling method associated to this poller.
+
+        :return: AnalyzeDocumentLROPollingMethod
+        :rtype: AnalyzeDocumentLROPollingMethod
+        """
+        return self._polling_method  # type: ignore
+
+    @property
+    def operation_id(self) -> str:
+        """Analyze operation result ID.
+
+        :rtype:str
+        """
+        return self.polling_method().operation_id
+
+
+class AnalyzeBatchDocumentsLROPollingMethod(LROBasePolling):
```

Comment:
It's unfortunate that we need to do a custom poller here just to handle a non-standard LRO status. Did the crew follow-up with the service team to get them to conform to the guidelines?